README.md

```diff
@@ -545,6 +546,55 @@ module "eks-jx" {
 
 :warning: **Note**: EKS node groups are supported in kubernetes v1.14+ and platform version eks.3
 
+### EBS CSI Driver
+In version 1.23 the Kubernetes in-tree to container storage interface (CSI) volume migration feature is enabled. This feature enables the replacement of existing Kubernetes in-tree storage plugins for Amazon EBS with a corresponding Amazon EBS CSI driver. If you use Amazon EBS volumes install the Amazon EBS CSI driver in your cluster before you update/create your cluster to/in version 1.23 [Kubernetes 1.23 Ref](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-1.23).
+
+An add-on is software that provides supporting operational capabilities to Kubernetes applications, but is not specific to the application. This includes software like observability agents or Kubernetes drivers that allow the cluster to interact with underlying AWS resources for networking, compute, and storage. [EKS Addons Guide](https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html)
+
+To enable the EBS CSI Driver (aws-ebs-csi-driver) set variable `enable_ebscsi_driver` to true. The version of the driver software is defined in the object variable `ebs_addon`
+```
+variable "ebs_addon" {
+  description = "EBS CSI driver addon"
+  type = object({
+    name    = string
+    version = string
+  })
+
+  default = {
+      name    = "aws-ebs-csi-driver"
+      version = "v1.21.0-eksbuild.1"
+    }
+}
+```
+To determine what versions of EBS CSI driver are supported use the command:
+```
+aws eks describe-addon-versions --addon-name "aws-ebs-csi-driver" | jq -r '.addons[].addonVersions[].addonVersion'
+```
+To display the latest version...
+```
+aws eks describe-addon-versions --addon-name "aws-ebs-csi-driver" | jq -r '.addons[0].addonVersions[].addonVersion'
+```
+Enabling the EBS CSI driver will append to the worker nodes the service role policy `AmazonEBSCSIDriverPolicy`
+```
+  workers_additional_policies = [
+    "arn:${data.aws_partition.current.partition}:iam::aws:policy/AmazonEC2ContainerRegistryPowerUser",
+    var.enable_ebscsi_driver ? "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy" : ""
+  ]
+```
+Setting the EKS add-ons is done through resource `aws_aks_addon`
+```
+resource "aws_eks_addon" "ebs_addon" {
+  count             = var.enable_ebscsi_driver ? 1 : 0
+  cluster_name      = var.cluster_name
+  addon_name        = var.ebs_addon.name
+  addon_version     = var.ebs_addon.version
+  resolve_conflicts = "OVERWRITE"
+  depends_on = [
+    null_resource.kubeconfig
+  ]
+}
+```
```

Comment:
I think this is only confusing to put in the documentation. A user that want to see the implementation can just look at the code.

Reply:
I agree. I can remove this.